vault_core/src/vault.rs

```diff
@@ -845,6 +847,106 @@ impl Vault {
         })
     }
 
+    /// Checks to see if the minimum amount out is acceptable
+    /// acceptable being defined as slippage being larger or equal to the minimum slippage
+    /// as well as accounting for withdraw fees
+    pub fn check_min_supported_mint_out(
```

Comment:
it seems like at first glance theres some overlap with burn_with_fee here. what do you think about:
- moving the main burn with fee logic to a separate method
- burn with fee calls this new method + does the modification to the vault at the end
- this method can use the new method

Reply:
Done